execution-plane/arcon/src/streaming/source/kafka.rs

```diff
@@ -1,41 +1,281 @@
+    }
 }
-impl<A: ArconType> Actor for KafkaSource<A> {
-    fn receive_local(&mut self, _sender: ActorRef, msg: &Any) {}
 
-    fn receive_message(&mut self, _sender: ActorPath, ser_id: u64, buf: &mut Buf) {}
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::{thread, time};
+    // Stub for window-results
+    #[arcon]
+    struct Thing {
+        id: u32,
+        attribute: i32,
+        location: Point,
+    }
+    #[arcon]
+    struct Point {
+        x: f32,
+        y: f32,
+    }
+    // JSON Example: {"id":1, "attribute":-13,"location":{"x":0.14124,"y":5882.231}}
+    mod sink {
+        use core::fmt::Debug;
+        use super::*;
+        use std::sync::Arc;
+
+        pub struct Sink<A: 'static + ArconType + Debug> {
+            ctx: ComponentContext<Sink<A>>,
+            pub result: Vec<ArconElement<A>>,
+            pub watermarks: Vec<Watermark>,
+        }
+        impl<A: ArconType + Debug> Sink<A> {
+            pub fn new() -> Sink<A> {
+                Sink {
+                    ctx: ComponentContext::new(),
+                    result: Vec::new(),
+                    watermarks: Vec::new(),
+                }
+            }
+        }
+        impl<A: ArconType + Debug> Provide<ControlPort> for Sink<A> {
+            fn handle(&mut self, _event: ControlEvent) -> () {}
+        }
+        impl<A: ArconType + Debug> Actor for Sink<A> {
+            fn receive_local(&mut self, _sender: ActorRef, msg: &Any) {
+                if let Some(event) = msg.downcast_ref::<ArconEvent<A>>() {
+                    match event {
+                        ArconEvent::Element(e) => {
+                            println!("Sink receieved element: {:?} // {:?}", e.data, e.timestamp);
+                            self.result.push(*e);
+                        }
+                        ArconEvent::Watermark(w) => {
+                            println!("Sink receieved watermark: {}", w.timestamp);
+                            self.watermarks.push(*w);
+                        }
+                    }
+                } else {
+                }
+            }
+            fn receive_message(&mut self, _sender: ActorPath, _ser_id: u64, _buf: &mut Buf) {}
+        }
+        impl<A: ArconType + Debug> ComponentDefinition for Sink<A> {
+            fn setup(&mut self, self_component: Arc<Component<Self>>) -> () {
+                self.ctx_mut().initialise(self_component);
+            }
+            fn execute(&mut self, _max_events: usize, skip: usize) -> ExecuteResult {
+                ExecuteResult::new(skip, skip)
+            }
+            fn ctx(&self) -> &ComponentContext<Self> {
+                &self.ctx
+            }
+            fn ctx_mut(&mut self) -> &mut ComponentContext<Self> {
+                &mut self.ctx
+            }
+            fn type_name() -> &'static str {
+                "EventTimeWindowAssigner"
+            }
+        }
+    }
+
+    #[test]
+    fn kafka_source() -> Result<()> { 
```

Comment:
This test does not look like it is doing anything?
And it is sleeping 10 seconds

Reply:
I was using the test to run "manual inspection tests" and figured it doesn't hurt to leave them as they were, useful for further development of them until we have a Kafka-mock or CI set-up with proper tests. It's not running when calling "cargo test" because it requires the feature flag to be added explicitly to the build/test command.